File: codes/ruby/chapter01/answer_A03.rb
# 入力
n, k = gets.split.map(&:to_i)
p = gets.split.map(&:to_i)
q = gets.split.map(&:to_i)

# 全探索（answer は「合計が k になる選び方が見つかったか」を示す）
answer = false
p.each do |x|
  q.each do |y|
    answer = true if x + y == k
  end
end

# 出力
if answer
  puts "Yes"
else
  puts "No"
end

File: codes/elixir/chapter01/answer_A03.exs
defmodule Solution do
  def main do
    k = String.trim(IO.gets("")) |> String.to_integer
    pn = IO.read(:line) |> String.split() |> Enum.map(&String.to_integer/1)
    qn = IO.read(:line) |> String.split() |> Enum.map(&String.to_integer/1)

    answer = check_sum_exists?(pn, qn, k)

    if answer do
      IO.puts "Yes"
    else
      IO.puts "No"
    end
  end

  defp check_sum_exists?([], _qn, _k), do: false
  defp check_sum_exists?([x | xs], qn, k) do
    if Enum.any?(qn, fn y -> x + y == k end) do
      true
    else
      check_sum_exists?(xs, qn, k)
    end
  end
end

Solution.main()
